sdk/identity/azure-identity/src/client_certificate_credential.cpp

```diff
@@ -399,16 +400,33 @@ ClientCertificateCredential::ClientCertificateCredential(
     CertificateThumbprint mdVec;
     try
     {
+      if (clientCertificatePath.empty())
+      {
+        throw AuthenticationException("Certificate file path is empty.");
+      }
+
+      using Azure::Core::_internal::StringExtensions;
+      std::string const PemExtension = ".pem";
+      auto const certFileExtensionStart = clientCertificatePath.find_last_of('.');
+      auto const certFileExtension = certFileExtensionStart != std::string::npos
+          ? clientCertificatePath.substr(certFileExtensionStart)
+          : std::string{};
```

Comment:
This isn't a very resilient test for getting a file extension (what happens if the filename or path/directory name has a period in it). We can't use C++ 17, but maybe we can borrow some inspiration from [std::filesystem::path::extension](https://en.cppreference.com/w/cpp/filesystem/path/extension)

Reply:
We search from the back of the string (`find_last_of()`), I think it is pretty resilient.
Feel free to log a work item to improve, maybe we need some facility in `Azure::Core` to provide the functionality similar to C++17 `std::filesystem`.